src/scanner.c

```diff
@@ -89,6 +91,40 @@ bool tree_sitter_scala_external_scanner_scan(void *payload, TSLexer *lexer,
       return false;
     }
 
+    if (valid_symbols[CATCH]) {
+      if (lexer->lookahead != 'c') return true;
+      advance(lexer);
+      if (lexer->lookahead != 'a') return true;
+      advance(lexer);
+      if (lexer->lookahead != 't') return true;
+      advance(lexer);
+      if (lexer->lookahead != 'c') return true;
+      advance(lexer);
+      if (lexer->lookahead != 'h') return true;
+      advance(lexer);
+      if (iswalpha(lexer->lookahead)) return true;
+      return false;
+    }
```

Comment:
Just checking - Is there ever a case where both `catch` and `finally` are valid at the same time?

Reply:
Do you mean valid interchangably?

Like:

```scala
try doTHing()
<here> <- both catch or finally are valid here
```

Is that way you mean?

Reply:
Yeah. So, after checking the grammar, I see that after the "body", both keywords `catch` and `finally` would be valid (because the catch clause is optional):

```
    try_expression: $ => prec.right(seq(
      'try',
      field('body', $._expression),
      optional($.catch_clause),
      optional($.finally_clause)
    )),
```

So I think that, if the keyword `catch` is valid, you still need to also check for the presence of the word `finally`.

Reply:
Right now, if `catch` is valid, and the lookahead is not the letter `c`, the scanner will return `true` (creating an automatic semicolon). This isn't quite right, because we still want to make sure that the next keyword isn't "finally".

Reply:
Ahhh ok, yea that totally makes sense. Alright, I accounted for that now.